extensions/app-runtime.md

```diff
@@ -0,0 +1,51 @@
+# Application Runtime Extensions
+
+This document specifies a contract between a platform and processes running in an application OCI image.  
+
+## Runtime Environment Variables
+
+The following environment variables extend the [Environment - Provided by the Platform](../buildpack.md#provided-by-the-platform) section in the Buildpack Interface Specification.
+The are exclusively available during the launch phase.
+
+| Env Variable         | Example Value                    | Description                       | Mandatory
+|----------------------|----------------------------------|----------------------------------------------
+| `PORT`               | `8080`                           | port for primary web process type | [x]
```

Comment:
I don’t think `PORT` or `CNB_APP_ROUTES` should be mandatory.  The task (i.e. non-LRP, not web-facing) use-case, should be a first-class citizen and in this use-case these concepts, routing of network traffic is an optional feature.  Think Batch processing where you’re running an application doing an ingest and transformation.

Reply:
What if `PORT` were mandatory for instances of the `web` process type?
Could `CNB_APP_ROUTES` be mandatory for app images at runtime, but empty (`{}`) when there are no web-based LRPs?

Trying to avoid adding an extra emptiness check to every simple web app that binds to `PORT`.

Reply:
After more thought, I wonder if these should be mandatory for "app process types" and not any specific process type identifier.

Reply:
I made all but `PORT` mandatory, in order to support worker processes. That said, I'm not convinced that this spec should apply to worker processes. E.g., do they really need an app name? Should we define a separate extension for worker LRPs?

Reply:
https://github.com/buildpack/spec/pull/42/commits/0f76577dc60ab6d06cfb9ca08600988288602d11

Reply:
I think the problem here is thinking that there is some differentiation between web and worker processes; this kind of thinking has really hamstrung CF to date.  If you take a step back and say that there are _processes_ and that some bind to ports and some don't, some live forever and some don't, and evaluate the requirements I think you'll see any route/port related information isn't a mandatory requirement.

I think buildpacks can build towards platform conventions (e.g. if my buildpack provides both a `web` and `task` process type, I'll only put `$PORT` in the `web` one) or application conventions (e.g. only if I notice that this uses Rails or Tomcat, I'll add `$PORT`) but I don't think this results in a mandatory, spec-level requirement.

Reply:
I don't have a strong opinion about keeping web and worker process types separate. I think https://github.com/buildpack/spec/commit/0f76577dc60ab6d06cfb9ca08600988288602d11 meets your requirements (note that `CNB_APP_ROUTES` can be `{}`, just want it to be valid JSON).